### ISSUE TITLE
No nb_uniq_visitors for Custom Date Range in Contents.getContentNames

### ISSUE THREAD

**Issue by cflee** [piwik/piwik#9089]
I've enabled "Unique Visitors" metric as stated in [FAQ](http://piwik.org/faq/how-to/faq_113/):

```
[General]
enable_processing_unique_visitors_year = 1
enable_processing_unique_visitors_range = 1
```

However, the `nb_uniq_visitors` value is still not available on the `Contents.getContentNames` API endpoint for `period` that is not `day`. Instead, only `sum_daily_nb_uniq_visitors` is available.

1. As `nb_uniq_visitors` is [defined](http://developer.piwik.org/api-reference/reporting-api#api-response-metric-definitions) as 'Number of unique visitors', can I confirm that since this is in context of Content Tracking, this is the number of unique visitors that have a "content impression"?
2. Is there any way of getting these values, for a custom date range:
  * number of unique visitors who have a "content impression"
  * number of unique visitors who have a "content interaction"

**Comment by mattab** [comment]
Do you still have the issue, for the new data @cflee ?

**Comment by cflee** [comment]
Yes, the issue is still there. I looked at the Archiver class but the aggregation seems to work very differently from the other plugins like Actions and Goals, so it's not affected by the `enable_processing_unique_visitors_range` setting.

Is this by design? I am trying to roll my own implementation of this metric externally, but I would definitely prefer if it is available in Piwik itself.

**Comment by mattab** [comment]
@cflee it's not really by design. ideally `enable_processing_unique_visitors_*` would consistently apply to Content Tracking and other plugins. If you are able to, we'd love to see a pull request to fix this behavior :-)

**Comment by rudyzeinoun** [comment]
Has a fix been made for this? I am trying to see unique visitors in a Custom Dimensions report, but that's also not available for ranges even when enabled in the settings. I can only find the sum, similar to @cflee's  problem.
